### ISSUE TITLE
Average out the denominator when computing focal loss

### ISSUE THREAD

**Issue by fredzzhang** [fredzzhang/spatio-attentive-graphs#39]
The focal loss used in the model is normalised by the number of positive logits, which tends to have unstable statistics. Therefore, it should be averaged across all sub-batches (ranks) to better utilise the large batch size.
